File: Tests/src/test/java/com/tle/webtests/pageobject/wizard/controls/EditBoxControl.java
package com.tle.webtests.pageobject.wizard.controls;

import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;

import com.tle.webtests.framework.PageContext;
import com.tle.webtests.pageobject.wizard.AbstractWizardControlPage;

public class EditBoxControl extends AbstractWizardControl<EditBoxControl>
{
	@FindBy(xpath = "id('{wizid}')//div[@class=\"input text\"]/input")
	private WebElement input;
	@FindBy(xpath = "id('{wizid}')//h3")
	private WebElement title;

	public EditBoxControl(PageContext context, int ctrlnum, AbstractWizardControlPage<?> page)
	{
		super(context, ctrlnum, page);
	}

	@Override
	protected WebElement findLoadedElement()
	{
		return input;
	}

	public boolean isEnabled()
	{
		return input.isEnabled();
	}

	public void setText(String text)
	{
		input.sendKeys(text);
	}

	public String getText()
	{
		return input.getAttribute("value");
	}

	public String getTitle()
	{
		return title.getText().trim();
	}
}

File: autotest/Tests/src/test/scala/equellatests/sections/wizard/EditBoxControl.scala
package equellatests.sections.wizard

import org.openqa.selenium.By

case class EditBoxControl(val page: WizardPageTab, val ctrlNum: Int) extends WizardControl {
  override def pageBy = By.xpath(s"id(${quoteXPath(ctrlId)})//div[@class='input text']/input")

  def value : String = pageElement.getAttribute("value")

  def value_=(v: String): Unit = {
    pageElement.clear()
    pageElement.sendKeys(v)
  }
}
